backend/internal/controller/server/server.go

```diff
@@ -21,7 +21,7 @@ func New(addr string) Server {
 	serv := server{
 		router:  mux.NewRouter(),
 		address: addr,
-		game:    model.Game{},
+		game:    model.NewGame(),
```

Comment:
Pass it in constructor, don't create it by yourself in server - it'll allow swapping implementations

Reply:
changed